libs/ngxtension/linked-query-param/src/linked-query-param.spec.ts

```diff
@@ -0,0 +1,247 @@
+			SearchComponent,
+		);
+		expect(instance.parseBehaviorWithParseAndNumberAttribute()).toBe(1);
+		await harness.navigateByUrl(
+			'/search?parseBehaviorWithParseAndNumberAttribute=2',
+		);
+		expect(instance.parseBehaviorWithParseAndNumberAttribute()).toBe(2);
+	}));
+
+	it('should parse and use the default value', fakeAsync(async () => {
+		const harness = await RouterTestingHarness.create();
+
+		const instance = await harness.navigateByUrl('/search', SearchComponent);
+		expect(instance.parseBehaviorWithDefault()).toBe(1); // default value
+
+		await harness.navigateByUrl('/search?parseBehaviorWithDefault=2');
+		expect(instance.parseBehaviorWithDefault()).toBe(2); // updated value
+
+		await harness.navigateByUrl('/search');
+		expect(instance.parseBehaviorWithDefault()).toBe(1); // default value
+	}));
+
+	it('should transform the value we set on the signal', fakeAsync(async () => {
+		const harness = await RouterTestingHarness.create();
+
+		const instance = await harness.navigateByUrl(
+			'/search?transformBehavior=value',
+			SearchComponent,
+		);
+		expect(instance.transformBehavior()).toBe('value');
+
+		instance.transformBehavior.set('value2');
+
+		// the value is not set on the signal itself, but only on the query param
+		expect(instance.transformBehavior()).toBe('value2');
+
+		tick();
+		expect(instance.route.snapshot.queryParams['transformBehavior']).toBe(
+			'VALUE2',
+		);
+	}));
+
+	it('should coalesce multiple set calls and only update the query param once', fakeAsync(async () => {
+		const harness = await RouterTestingHarness.create();
+		const instance = await harness.navigateByUrl('/search', SearchComponent);
+
+		instance.defaultBehavior.set('value');
+		expect(instance.defaultBehavior()).toBe('value');
+
+		instance.defaultBehavior.set('value2');
+		expect(instance.defaultBehavior()).toBe('value2');
+
+		// TODO: find a better way to test this
+
+		expect(instance.route.snapshot.queryParams['defaultBehavior']).toBe(
+			undefined,
+		);
+
+		tick();
+		expect(instance.route.snapshot.queryParams['defaultBehavior']).toBe(
+			'value2',
+		);
+	}));
+
+	it('should handle null values', fakeAsync(async () => {
+		const harness = await RouterTestingHarness.create();
+		const instance = await harness.navigateByUrl('/search', SearchComponent);
+
+		instance.defaultBehavior.set(null);
+		expect(instance.defaultBehavior()).toBe(null);
+
+		tick();
+		expect(instance.route.snapshot.queryParams['defaultBehavior']).toBe(
+			undefined,
+		);
+
+		await harness.navigateByUrl('/search?defaultBehaviorWithDefault=value1');
+		expect(instance.defaultBehaviorWithDefault()).toBe('value1');
+
+		// @ts-ignore
+		instance.defaultBehaviorWithDefault.set(null);
+
+		expect(instance.defaultBehaviorWithDefault()).toBe('value'); //THINK: should this be null?
+
+		tick();
+		expect(
+			instance.route.snapshot.queryParams['defaultBehaviorWithDefault'],
+		).toBe(undefined);
+	}));
+
+	//
```

Comment:
TODO: also test what happens when we update a query param and some other linkedQueryParam is listening to the same thing, test the timing of that and try to fix it.

Reply:
I got this working in the latest version. So adding a test is the only thing left.